### ISSUE TITLE
Changing default method to static should be separate case

### ISSUE THREAD

**Issue by siom79** [siom79/japicmp#363]
see discussion in https://github.com/siom79/japicmp/commit/a941c343cfddf2dbcb19b2d0281c1e6026e07b69#r128449820

**Comment by Marcono1234** [comment]
From https://github.com/siom79/japicmp/commit/a941c343cfddf2dbcb19b2d0281c1e6026e07b69#r128486069:
> In case it was abstract and changed to static, then the Class C must have an implementation of the method (as it was abstract) and therefore it is still source compatible (`new C().doSomething()` calls the method in `C`) and binary compatible (the method in `C` gets called).

The problem here is though when the method is not called on the class `C` but on a variable with the type of the interface `I`:
```
I i = ...;
i.doSomething();
```
If `doSomething()` is changed from abstract to static this causes
- Source incompatibility:
  ```
  illegal static interface method call
  ```
- Binary incompatibility:
  ```
  IncompatibleClassChangeError: Expected instance not static method 'void I.doSomething()'
  ```

**Comment by siom79** [comment]
@Marcono1234 Got it. Thanks for pointing out.

**Comment by Marcono1234** [comment]
Thanks for the changes!

`METHOD_NON_STATIC_IN_INTERFACE_NOW_STATIC` has still `JApiSemanticVersionLevel.MINOR` though. Is that intentional or should it rather be `MAJOR` now?

**Comment by siom79** [comment]
Yes, you are right again. :smile: And then we don't need `METHOD_DEFAULT_NOW_STATIC`, too.

**Comment by siom79** [comment]
Released with 0.18.1
